### PR TITLE
Support Pydantic v2

### LINKED ISSUE #74: Incompatible with pydantic 2.0.1 - Error validation with GET
```
class SampleModel(BaseModel):
    modelcode: str

@personalizations_blueprint.route('/data', methods=['GET'])
@jwt_required()
@validate()
def getData(query: SampleModel):
   .....


```

Validating against this model produces the following error:

AttributeError: 'FieldInfo' object has no attribute 'is_complex'

 I'm using pydantic 2.0.1 with Flask-Pydantic 0.11.0

After a bit checking, I've found that downgrading pydantic to version 1 works. So I think you need to update your support for pydantic version 2.

### PR DESCRIPTION
**Pull request by jkseppan**
Fixes #74

### CODE REVIEW

File: example_app/app.py [hunk 1]
```diff
@@ -60,7 +60,7 @@ def post():
 
 @app.route("/form", methods=["POST"])
 @validate(form=FormModel, query=QueryModel)
-def post():
+def post2():
```
Comment (adeadman) [inline example_app/app.py:63]: Worth naming as `form_post` or similar, based on the route used?
```suggestion
def form_post():
```

File: example_app/app.py [hunk 2]
```diff
@@ -90,7 +90,7 @@ def post_kwargs(body: BodyModel, query: QueryModel):
 
 @app.route("/form/kwargs", methods=["POST"])
 @validate()
-def post_kwargs(form: FormModel, query: QueryModel):
+def post_kwargs2(form: FormModel, query: QueryModel):
```
Comment (adeadman) [inline example_app/app.py:93]:
```suggestion
def form_post_kwargs(form: FormModel, query: QueryModel):
```

File: example_app/example.py [hunk 3]
```diff
@@ -92,7 +92,7 @@ def post(body: RequestBodyModel):
 
 @app.route("/form", methods=["POST"])
 @validate()
-def post(form: FormModel):
+def post2(form: FormModel):
```
Comment (adeadman) [inline example_app/example.py:95]:
```suggestion
def form_post(form: FormModel):
```

**Review by adeadman — COMMENTED** [review]
Thanks for this, it works well. In addition to a code owner bump, is it worth updating the example function names from `post2` to e.g. `form_post` for improved clarity? In any case it all LGTM.

**Comment by ronyriachy** [comment]
When will this be merged?

**Comment by Matvii-Boichenko** [comment]
@adeadman sorry to disturb you. It seems this PR with linda critical fix is stuck in this state for more than a month. Could you please provide info about the cause? It seems workflow just stopped. Is there any way to finish it or some info when it may happen automatically?

**Comment by adeadman** [comment]
> @adeadman sorry to disturb you. It seems this PR with kinda critical fix is stuck in this state for more than a month. Could you please provide info about the cause? It seems workflow just stopped. Is there any way to finish it or some info when it may happen automatically?

I am not the maintainer of this library. I think you mean @bauerji, who would need to look at the build statuses and approve the merge. Looking at their Github profile, they have not been active for a number of months and I haven't found any more publicly available information about the status of this project or the maintainer. Hope this helps

**Comment by jim-schwoebel** [comment]
Bumping this as well! Pydantic v2 support would be great.

**Comment by mjgp2** [comment]
I think this may end up needing to be forked, the owner seems MIA from Github unfortunately, hopefully he is OK.

**Comment by afallou** [comment]
FYI I emailed the repo author on the email address listed on PyPI to tell him I can take over the project. Hopefully that way we can get a clean transfer of the repo and pypi name ownership, rather than a fork.

**Comment by amjed-ali-k** [comment]
@bauerji Kindly look into this

**Comment by jim-schwoebel** [comment]
Anyone down to fork and maintain in public pypi?

Maybe flask-pydantic2 or something as a name

**Comment by bauerji** [comment]
Hey all, thank you for your contribution.

Sadly, I've had no time for this project recently.

Is there anyone who would be willing to take over? I think it would be better to continue in this rather than forking etc.

**Comment by amjed-ali-k** [comment]
Kindly merge this and release.

**Comment by bauerji** [comment]
@jkseppan, I am sorry for such a late response. Thank you for your contribution.

Could you please adjust the tests a bit to pass? It would be nice if we could get rid of the pydantic version in error messages, so that we don't have to adjust it every time.

See the test results above for more details.

**Comment by jkseppan** [comment]
Looks like there are a few reasons for the test failures. I can make the tests pass locally by downgrading flask to 2.3.3, pydantic to 2.1 and pytest-flask to 1.2.0. The pydantic version number problem should be easy enough to fix, no idea yet about flask 3.0 changes.

**Comment by bauerji** [comment]
@jkseppan please, check https://github.com/bauerji/flask-pydantic/pull/77 there are some changes that might help you with that

**Comment by jkseppan** [comment]
@bauerji The other checks are passing, I don't know what's up with the two that are stuck.

**Comment by bauerji** [comment]
@jkseppan Thank you very much!

Released in a new version `0.12.0` 🎉

**Pull request merged**